config/reader.go

```diff
@@ -427,7 +437,8 @@ func processConfig(unprocessedConfig *Config, fromCloud bool, logger logging.Log
 
 	// We keep track of resource configs per API to facilitate linking resource configs to
 	// its associated resource configs. Associated resource configs are configs that are
-	// linked to and used by a different resource config.
+	// linked to and used by a different resource config. Within the rdk repo, the only API
+	// that uses associated resource configs is the data manager service.
```

Comment:
```suggestion
	// linked to and used by a different resource config. See the data manager 
	// service for an example of a resource that uses associated resource configs.
```

"the only API" feels like something could easily change, so I'd prefer to avoid making this claim in a code comment. I do like how we are giving a concrete example of how this is being used tho :+1: